src/V3RandomizeMethod.cpp

```diff
@@ -0,0 +1,234 @@
+                                        AstMemberDType* memberp = nullptr) {
+        AstStructDType* structDtp = VN_CAST(
+            memberp ? memberp->subDTypep()->skipRefp() : varp->dtypep()->skipRefp(), StructDType);
+        if (structDtp) {
+            AstNodeMath* randp = nullptr;
+            offset += memberp ? memberp->lsb() : 0;
+            for (auto* memberp = structDtp->membersp(); memberp;
+                 memberp = VN_CAST(memberp->nextp(), MemberDType))
+                randp = !randp ? newStdRandomize(fl, varp, offset, memberp)
+                               : new AstAnd(
+                                   fl, randp,
+                                   newStdRandomize(fl, varp->cloneTree(false), offset, memberp));
+            return randp;
+        } else {
+            return new AstStdRandomize(fl, varp, offset, memberp);
+        }
+    }
+
+    // VISITORS
+    virtual void visit(AstClass* nodep) override {
+        iterateChildren(nodep);
+        if (!nodep->user1()) return;  // Doesn't need randomize, or already processed
+        UINFO(9, "Define randomize() for " << nodep << endl);
+        auto* funcp = V3RandomizeMethod::newRandomizeFunc(nodep);
+        auto* fvarp = VN_CAST(funcp->fvarp(), Var);
+        funcp->addStmtsp(new AstAssign(
+            nodep->fileline(), new AstVarRef(nodep->fileline(), fvarp, VAccess::WRITE),
+            new AstConst(nodep->fileline(), AstConst::WidthedValue(), 32, 1)));
+        auto* classp = nodep;
+        for (auto* classp = nodep; classp;
+             classp = classp->extendsp() ? classp->extendsp()->classp() : nullptr) {
+            for (auto* memberp = classp->stmtsp(); memberp; memberp = memberp->nextp()) {
+                auto* memberVarp = VN_CAST(memberp, Var);
+                if (!memberVarp || !memberVarp->isRand()) continue;
+                AstNode* stmtp = nullptr;
+                if (VN_IS(memberp->dtypep()->skipRefp(), BasicDType)
+                    || VN_IS(memberp->dtypep()->skipRefp(), StructDType)) {
+                    auto* refp = new AstVarRef(nodep->fileline(), memberVarp, VAccess::WRITE);
+                    stmtp = newStdRandomize(nodep->fileline(), refp);
+                } else if (auto* classRefp = VN_CAST(memberp->dtypep(), ClassRefDType)) {
+                    auto* refp = new AstVarRef(nodep->fileline(), memberVarp, VAccess::WRITE);
+                    auto* funcp = V3RandomizeMethod::newRandomizeFunc(classRefp->classp());
+                    auto* callp = new AstMethodCall(nodep->fileline(), refp, "randomize", nullptr);
+                    callp->taskp(funcp);
+                    callp->dtypeFrom(funcp);
+                    stmtp = callp;
+                } else {
+                    memberp->v3warn(E_UNSUPPORTED,
+                                    "Unsupported: random member variables with type "
+                                        << memberp->dtypep()->skipRefp()->type().ascii());
```

Comment:
```suggestion
                                        << memberp->dtypep()->prettyDTypeNameQ());
```

Reply:
I think I used `ascii` because it says something like "array" or "associative array" as it communicates more clearly what is unsupported, instead of something like `int[5]`. But I can change it if you want.

Reply:
We should use pretty... as that's what's used elsewhere.  If we want to separately improve pretty... to make nicer names in another pull I'm all for it, it's probably unintntional if ascii() looks better.